### ISSUE TITLE
MyPy raising 'unrecognized arguments' error for changed files

### ISSUE THREAD

**Issue by pgergov** [pre-commit/mirrors-mypy#26]
Hello,

I have a Django project and I'm trying to add a MyPy pre-commit hook.
When adding MyPy to the pre-commit config yaml and trying to commit the following error occurs `mypy: error: unrecognized arguments: apps/events/admin.py apps/events/models.py`:

```bash
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

usage: mypy [-h] [-v] [-V] [more options; see below]
            [-m MODULE] [-p PACKAGE] [-c PROGRAM_TEXT] [files ...]
mypy: error: unrecognized arguments: apps/events/admin.py apps/events/models.py
```

Here's how the yaml configuration looks like:

```yaml
-   repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.770
    hooks:
    -   id: mypy
        args: [--python-version 3.8, --warn-unused-configs, --ignore-missing-imports, --pretty]
        additional_dependencies: [django-stubs==1.5.0]
```

When running mypy locally with the same settings everything works fine:

```bash
mypy --python-version 3.8 --warn-unused-configs --ignore-missing-imports --pretty apps/events/admin.py apps/events/models.py
Success: no issues found in 2 source files
```
Here're my local dependencies:

```bash
pip3 freeze | grep 'pre-commit\|mypy\|django-stubs\|Django'
Django==3.1.1
django-stubs==1.5.0
mypy==0.770
mypy-extensions==0.4.3
pre-commit==2.7.1
```

---

A couple of things I tried (not sure if they help, but I'll list them anyway):

* Using the latest tag `v0.782` in the yaml configuration, but got the same error.
* Running `pre-commit run --all-files` and I ended up with the same error `mypy: error: unrecognized arguments:` for all files.
* Looking at this https://github.com/pre-commit/mirrors-mypy/blob/master/.pre-commit-hooks.yaml I tried passing the arguments in double quotes, but got the same error:

```yaml
-   repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.770
    hooks:
    -   id: mypy
        args: ["--python-version 3.8", "--warn-unused-configs", "--ignore-missing-imports", "--pretty"]
        additional_dependencies: [django-stubs==1.5.0]
```

I checked all issues in the repo and googled about mine, but I couldn't find an answer. If no one else has this particular problem I guess I'm doing something wrong. Any help is highly appreciated :beers:

**Comment by asottile** [comment]
you're a little incorrect on the equivalent command, this is the equivalent command to what you've configured:

```bash
mypy "--python-version 3.8" --warn-unused-configs --scripts-are-modules --pretty t.py
```

I take it you can figure out the problem from there -- comment back if you can't :)

**Comment by pgergov** [comment]
@asottile Got it :+1: Thank you very much for looking into this!

Just a follow-up comment if someone else comes across this issue.  The problem was with the way I was passing the `python-version` argument. Here's the correct way:

```yaml
args: [--python-version=3.8, --warn-unused-configs, --ignore-missing-imports, --pretty]
```
